editor/editor_resource_picker.cpp

```diff
@@ -842,8 +848,8 @@ void EditorResourcePicker::set_base_type(const String &p_base_type) {
 	// There is a possibility that the new base type is conflicting with the existing value.
 	// Keep the value, but warn the user that there is a potential mistake.
 	if (!base_type.is_empty() && edited_resource.is_valid()) {
-		HashSet<StringName> allowed_types;
-		_get_allowed_types(true, &allowed_types);
+		_ensure_allowed_types();
+		HashSet<StringName> allowed_types = allowed_types_with_convert;
```

Comment:
Should we force updating the allowed types even if already cached, when changing `base_type`?

Like `_ensure_allowed_types(bool p_force)` that would bypass the `!allowed_types_without_convert.is_empty()` early return.

I'm not too familiar with this code but wouldn't this alleviate the issue with possibly requiring an editor restart to take new allowed types into account?

Reply:
Editor restart is not required, only inspector refresh. When global classes change, all resource pickers already in scene tree will have outdated information and need to be recreated. Inspector pickers are recreated with every object edit, not sure if there are some that would stay outdated (maybe in e.g. project settings).

Also base type is set once per picker, idk if there are exceptions.